cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
```

Comment:
Shouldn't the features be associated with a particular input tensor?

I would expect something like `def fprop(self, x)` that returns a dictionary mapping layer names to activations computed on `x`.